In [1]:
#Import Library
library(tidyverse)

── Attaching packages ─────────────────────────────────────── tidyverse 1.2.1 ──
✔ ggplot2 3.1.0     ✔ purrr   0.2.5
✔ tibble  2.0.1     ✔ dplyr   0.7.6
✔ tidyr   0.8.1     ✔ stringr 1.3.1
✔ readr   1.1.1     ✔ forcats 0.3.0
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()


In [2]:
# Converts all factors to character class
unfactorize <- function(df){
  for(i in which(sapply(df, class) == "factor")) df[[i]] = as.character(df[[i]])
  return(df)
}
# Code from user "By0" at https://stackoverflow.com/questions/2851015/convert-data-frame-columns-from-factors-to-characters (line 14)

In [3]:
# Get and view last meta dataset
letters  <- unfactorize(read.csv("20201118_AM_Meta2Merge.csv"))
glimpse(letters)

Observations: 1,212
Variables: 32
$ X                       <int> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14…
$ docsequence             <int> 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 3…
$ docid                   <chr> "S10003-D023", "S10003-D024", "S10003-D025", …
$ docyear                 <int> 1836, 1836, 1837, 1837, 1838, 1838, 1838, 183…
$ doctype                 <chr> "Letter", "Letter", "Letter", "Letter", "Lett…
$ allsubject              <chr> "Childbirth; Church attendance; Cities; Farms…
$ broadsubj               <chr> "Health; Religion; Communities; Relationships…
$ personalevent           <chr> NA, NA, NA, NA, NA, NA, NA, NA, "Physical ill…
$ wwritten                <chr> "Baltimore, MD; Maryland; United States; Mid-…
$ docauthorid             <chr> "per0022938", "per0022938", "per0022938", "pe…
$ docauthorname           <chr> "Bruns, Jette, 1813-1899", "Bruns, Jette, 181…
$ language                <chr> "English; German", "English; German", "Englis…
$ editor          

In [5]:
# How many letters were written before 1914
nrow(subset(letters, docyear < 1914))

# How many doc years are unknown?
sum(is.na(letters$docyear))

[1] 917

[1] 20

In [5]:
# Make subset of letters writte before 1914 or for whom docyear is NA.
lettersNew  <- subset(letters, docyear < 1914 | is.na(letters$docyear))
nrow(lettersNew)

[1] 937

In [6]:
# Who are the NAs and when did they live (contained in docauthorid)
table(lettersNew$docauthorname[is.na(lettersNew$docyear)])


             Beltman, H., fl. 1878        Birkbeck, Morris, 1764-1825 
                                 1                                  1 
Bishop, Robert Hamilton, 1777-1855                     Brinks, Roelof 
                                 1                                  1 
        De Jong, William, fl. 1888          Delfino, Diego, 1874-1926 
                                 2                                  2 
            Dunnink, Mr., fl. 1850                      Johnson, Nils 
                                 1                                  1 
                    Karsten, J. H.                        Koopman, H. 
                                 1                                  1 
                      Kroes, Henry            Lankester, P., fl. 1849 
                                 1                                  2 
                Nywening, U. V. L.                  Posthumus, Wietze 
                                 1                                  1 
     

<b>Decision</b>: Drop Diego and authors without any year associated. Note: <i>Fl.</i> means flourished and is understood to mark a person's active years. People whose active year is 1850 or earlier will be included. Dates afterward excluded. Get those docauthorids.

In [7]:
# Put names in list
names = c("Beltman, H., fl. 1878",
          "De Jong, William, fl. 1888", 
          "Karsten, J. H.", 
          "Kroes, Henry", 
          "Nywening, U. V. L.",
          "Sneller, Matje",
          "Veldhuis, Zwiertje",
          "Brinks, Roelof",
          "Delfino, Diego, 1874-1926", 
          "Johnson, Nils",
          "Koopman, H.",
          "Posthumus, Wietze",
          "Te Selle, Harm, fl. 1865")

In [8]:
# Do some of these authors have letters with docyear provided?
lettersNew[lettersNew$docauthorname %in% names, c("docauthorname", "docid", "docyear")]

docauthorname              docid       docyear
413  Delfino, Diego, 1874-1926  S12296-D002 1912   
418  Delfino, Diego, 1874-1926  S12296-D008   NA   
419  Delfino, Diego, 1874-1926  S12296-D009   NA   
933  Sneller, Matje             S9831-D008    NA   
935  Brinks, Roelof             S9831-D010    NA   
936  Veldhuis, Zwiertje         S9831-D014    NA   
946  Kroes, Henry               S9831-D026    NA   
950  Nywening, U. V. L.         S9831-D036    NA   
951  De Jong, William, fl. 1888 S9831-D037    NA   
952  Posthumus, Wietze          S9831-D038    NA   
953  De Jong, William, fl. 1888 S9831-D039    NA   
954  Koopman, H.                S9831-D040    NA   
960  Karsten, J. H.             S9831-D047    NA   
965  Beltman, H., fl. 1878      S9831-D055    NA   
970  Te Selle, Harm, fl. 1865   S9831-D061    NA   
971  Te Selle, Harm, fl. 1865   S9831-D062  1870   
972  Te Selle, Harm, fl. 1865   S9831-D063  1882   
1166 Johnson, Nils              S9983-D055    NA

Include the people who whose other letters predate 1914.

In [9]:
# Revising the list above.
names = c("Beltman, H., fl. 1878",
          "De Jong, William, fl. 1888", 
          "Karsten, J. H.", 
          "Kroes, Henry", 
          "Nywening, U. V. L.",
          "Sneller, Matje",
          "Veldhuis, Zwiertje",
          "Brinks, Roelof",
          "Johnson, Nils",
          "Koopman, H.",
          "Posthumus, Wietze")

# Remove cases with the docauthors above from the dataset.
lettersFinal  <- subset(lettersNew, !(docauthorname %in% names))

# Who are the docyear-NAs that were saved?
table(lettersFinal$docauthorname[is.na(lettersFinal$docyear)])


       Birkbeck, Morris, 1764-1825 Bishop, Robert Hamilton, 1777-1855 
                                 1                                  1 
         Delfino, Diego, 1874-1926             Dunnink, Mr., fl. 1850 
                                 2                                  1 
           Lankester, P., fl. 1849           Te Selle, Harm, fl. 1865 
                                 2                                  1 

In [40]:
glimpse(lettersFinal)

Observations: 925
Variables: 32
$ X                       <int> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14…
$ docsequence             <int> 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 3…
$ docid                   <chr> "S10003-D023", "S10003-D024", "S10003-D025", …
$ docyear                 <int> 1836, 1836, 1837, 1837, 1838, 1838, 1838, 183…
$ doctype                 <chr> "Letter", "Letter", "Letter", "Letter", "Lett…
$ allsubject              <chr> "Childbirth; Church attendance; Cities; Farms…
$ broadsubj               <chr> "Health; Religion; Communities; Relationships…
$ personalevent           <chr> NA, NA, NA, NA, NA, NA, NA, NA, "Physical ill…
$ wwritten                <chr> "Baltimore, MD; Maryland; United States; Mid-…
$ docauthorid             <chr> "per0022938", "per0022938", "per0022938", "pe…
$ docauthorname           <chr> "Bruns, Jette, 1813-1899", "Bruns, Jette, 181…
$ language                <chr> "English; German", "English; German", "Englis…
$ editor            

In [26]:
# How many IDs were removed from the dataset and what are they
originalIDs  <- letters$docid
finalIDs <- lettersFinal$docid
removedIDs <- originalIDs[!(originalIDs %in% finalIDs)]
length(removedIDs)
removedIDs

[1] 287

[1] "S11067-D063" "S11067-D064" "S11067-D065" "S11067-D066" "S11067-D067"
  [6] "S11067-D072" "S11067-D073" "S11067-D080" "S11067-D082" "S11067-D086"
 [11] "S11067-D087" "S11067-D089" "S11067-D092" "S11067-D095" "S11067-D100"
 [16] "S11067-D101" "S11067-D102" "S11067-D104" "S11067-D109" "S11067-D111"
 [21] "S11067-D113" "S11068-D008" "S11068-D016" "S11068-D025" "S11068-D034"
 [26] "S11068-D040" "S11068-D043" "S11068-D044" "S11068-D045" "S11068-D046"
 [31] "S11068-D047" "S11068-D048" "S11068-D057" "S11068-D077" "S11068-D078"
 [36] "S11068-D083" "S11068-D085" "S11068-D101" "S11068-D104" "S11073-D008"
 [41] "S11073-D009" "S11073-D010" "S11073-D011" "S11073-D012" "S11073-D013"
 [46] "S11073-D014" "S11073-D016" "S11073-D018" "S11073-D019" "S11073-D020"
 [51] "S11073-D022" "S11073-D023" "S11073-D024" "S11073-D026" "S11073-D029"
 [56] "S11073-D030" "S11073-D031" "S11073-D032" "S11073-D034" "S11073-D035"
 [61] "S11073-D036" "S11073-D037" "S11073-D038" "S11073-D039" "S11073-D041"
 [66] "S11073-D043" "S11073-D044" "S11073-D045" "S11073-D047" "S11073-D048"
 [71] "S11073-D050" "S11073-D051" "S11073-D053" "S11073-D054" "S11073-D056"
 [76] "S11073-D057" "S11073-D059" "S11073-D061" "S11073-D063" "S11073-D064"
 [81] "S11073-D065" "S11073-D067" "S11073-D068" "S11073-D069" "S11073-D070"
 [86] "S11073-D071" "S11073-D072" "S11073-D073" "S11073-D074" "S11073-D076"
 [91] "S11073-D078" "S11073-D079" "S11073-D080" "S11073-D083" "S11073-D084"
 [96] "S11073-D085" "S11073-D086" "S11073-D087" "S11073-D088" "S11073-D090"
[101] "S11073-D091" "S11073-D093" "S11073-D094" "S11073-D095" "S11073-D097"
[106] "S11073-D099" "S11073-D100" "S11073-D101" "S11073-D103" "S11073-D104"
[111] "S11073-D105" "S11073-D107" "S11073-D109" "S11073-D110" "S11073-D112"
[116] "S11073-D113" "S11073-D116" "S11073-D117" "S11073-D119" "S11073-D120"
[121] "S11073-D122" "S11073-D123" "S11073-D125" "S11073-D126" "S11073-D127"
[126] "S11073-D128" "S11073-D130" "S11073-D131" "S11073-D133" "S11073-D134"
[131] "S11073-D135" "S11073-D137" "S11073-D138" "S11073-D139" "S11073-D140"
[136] "S11073-D142" "S11073-D143" "S11073-D144" "S11073-D145" "S11073-D146"
[141] "S11073-D147" "S11073-D148" "S11073-D149" "S11073-D150" "S11073-D152"
[146] "S11073-D153" "S11073-D154" "S11073-D155" "S11073-D156" "S11073-D157"
[151] "S11073-D158" "S11073-D159" "S11073-D160" "S11073-D161" "S11073-D162"
[156] "S11073-D163" "S11073-D164" "S11073-D165" "S11073-D166" "S11073-D167"
[161] "S11073-D168" "S11073-D169" "S11073-D170" "S11073-D171" "S11073-D172"
[166] "S11073-D173" "S11073-D174" "S11073-D176" "S11073-D177" "S11073-D178"
[171] "S11073-D179" "S11073-D180" "S11073-D182" "S11073-D183" "S11073-D184"
[176] "S11073-D185" "S11073-D186" "S11073-D187" "S11073-D188" "S11073-D189"
[181] "S11073-D190" "S11073-D191" "S11073-D192" "S11073-D193" "S11073-D194"
[186] "S11073-D195" "S11073-D196" "S12296-D003" "S12296-D005" "S12296-D006"
[191] "S12296-D007" "S12296-D010" "S12296-D011" "S12296-D012" "S12296-D013"
[196] "S12296-D014" "S12296-D015" "S12296-D016" "S12296-D017" "S12296-D018"
[201] "S12296-D019" "S12296-D020" "S12296-D021" "S12296-D022" "S12296-D023"
[206] "S12296-D024" "S12296-D025" "S12296-D026" "S12296-D027" "S12296-D028"
[211] "S12296-D029" "S12296-D030" "S12296-D031" "S12296-D032" "S12296-D033"
[216] "S12296-D034" "S12296-D035" "S12296-D036" "S12296-D037" "S12296-D038"
[221] "S12296-D039" "S12296-D040" "S12296-D041" "S12296-D042" "S12296-D043"
[226] "S12296-D044" "S12296-D045" "S12296-D046" "S12296-D047" "S12296-D048"
[231] "S12296-D049" "S12296-D050" "S12296-D051" "S12296-D052" "S12296-D053"
[236] "S12296-D054" "S12296-D055" "S12296-D056" "S12296-D058" "S12296-D059"
[241] "S12296-D060" "S2407-D011"  "S2407-D012"  "S2407-D013"  "S530-D007"  
[246] "S530-D008"   "S530-D009"   "S530-D010"   "S530-D011"   "S530-D012"  
[251] "S530-D013"   "S530-D016"   "S530-D017"   "S530-D018"   "S530-D019"  
[256] "S7560-D013"  "S7560-D014"  "S7560-D015"  "S7560-D016"  "S7560-D017" 
[261] "S7560-D018"  "S7560-D019"  "S9645-D024"  "S

This adds up nicely. Now I need to remove these files from the folder.

In [27]:
# .txt to all IDs
fileids <- paste(removedIDs, ".txt", sep="")
fileids

[1] "S11067-D063.txt" "S11067-D064.txt" "S11067-D065.txt" "S11067-D066.txt"
  [5] "S11067-D067.txt" "S11067-D072.txt" "S11067-D073.txt" "S11067-D080.txt"
  [9] "S11067-D082.txt" "S11067-D086.txt" "S11067-D087.txt" "S11067-D089.txt"
 [13] "S11067-D092.txt" "S11067-D095.txt" "S11067-D100.txt" "S11067-D101.txt"
 [17] "S11067-D102.txt" "S11067-D104.txt" "S11067-D109.txt" "S11067-D111.txt"
 [21] "S11067-D113.txt" "S11068-D008.txt" "S11068-D016.txt" "S11068-D025.txt"
 [25] "S11068-D034.txt" "S11068-D040.txt" "S11068-D043.txt" "S11068-D044.txt"
 [29] "S11068-D045.txt" "S11068-D046.txt" "S11068-D047.txt" "S11068-D048.txt"
 [33] "S11068-D057.txt" "S11068-D077.txt" "S11068-D078.txt" "S11068-D083.txt"
 [37] "S11068-D085.txt" "S11068-D101.txt" "S11068-D104.txt" "S11073-D008.txt"
 [41] "S11073-D009.txt" "S11073-D010.txt" "S11073-D011.txt" "S11073-D012.txt"
 [45] "S11073-D013.txt" "S11073-D014.txt" "S11073-D016.txt" "S11073-D018.txt"
 [49] "S11073-D019.txt" "S11073-D020.txt" "S11073-D022.txt" "S11073-D023.txt"
 [53] "S11073-D024.txt" "S11073-D026.txt" "S11073-D029.txt" "S11073-D030.txt"
 [57] "S11073-D031.txt" "S11073-D032.txt" "S11073-D034.txt" "S11073-D035.txt"
 [61] "S11073-D036.txt" "S11073-D037.txt" "S11073-D038.txt" "S11073-D039.txt"
 [65] "S11073-D041.txt" "S11073-D043.txt" "S11073-D044.txt" "S11073-D045.txt"
 [69] "S11073-D047.txt" "S11073-D048.txt" "S11073-D050.txt" "S11073-D051.txt"
 [73] "S11073-D053.txt" "S11073-D054.txt" "S11073-D056.txt" "S11073-D057.txt"
 [77] "S11073-D059.txt" "S11073-D061.txt" "S11073-D063.txt" "S11073-D064.txt"
 [81] "S11073-D065.txt" "S11073-D067.txt" "S11073-D068.txt" "S11073-D069.txt"
 [85] "S11073-D070.txt" "S11073-D071.txt" "S11073-D072.txt" "S11073-D073.txt"
 [89] "S11073-D074.txt" "S11073-D076.txt" "S11073-D078.txt" "S11073-D079.txt"
 [93] "S11073-D080.txt" "S11073-D083.txt" "S11073-D084.txt" "S11073-D085.txt"
 [97] "S11073-D086.txt" "S11073-D087.txt" "S11073-D088.txt" "S11073-D090.txt"
[101] "S11073-D091.txt" "S11073-D093.txt" "S11073-D094.txt" "S11073-D095.txt"
[105] "S11073-D097.txt" "S11073-D099.txt" "S11073-D100.txt" "S11073-D101.txt"
[109] "S11073-D103.txt" "S11073-D104.txt" "S11073-D105.txt" "S11073-D107.txt"
[113] "S11073-D109.txt" "S11073-D110.txt" "S11073-D112.txt" "S11073-D113.txt"
[117] "S11073-D116.txt" "S11073-D117.txt" "S11073-D119.txt" "S11073-D120.txt"
[121] "S11073-D122.txt" "S11073-D123.txt" "S11073-D125.txt" "S11073-D126.txt"
[125] "S11073-D127.txt" "S11073-D128.txt" "S11073-D130.txt" "S11073-D131.txt"
[129] "S11073-D133.txt" "S11073-D134.txt" "S11073-D135.txt" "S11073-D137.txt"
[133] "S11073-D138.txt" "S11073-D139.txt" "S11073-D140.txt" "S11073-D142.txt"
[137] "S11073-D143.txt" "S11073-D144.txt" "S11073-D145.txt" "S11073-D146.txt"
[141] "S11073-D147.txt" "S11073-D148.txt" "S11073-D149.txt" "S11073-D150.txt"
[145] "S11073-D152.txt" "S11073-D153.txt" "S11073-D154.txt" "S11073-D155.txt"
[149] "S11073-D156.txt" "S11073-D157.txt" "S11073-D158.txt" "S11073-D159.txt"
[153] "S11073-D160.txt" "S11073-D161.txt" "S11073-D162.txt" "S11073-D163.txt"
[157] "S11073-D164.txt" "S11073-D165.txt" "S11073-D166.txt" "S11073-D167.txt"
[161] "S11073-D168.txt" "S11073-D169.txt" "S11073-D170.txt" "S11073-D171.txt"
[165] "S11073-D172.txt" "S11073-D173.txt" "S11073-D174.txt" "S11073-D176.txt"
[169] "S11073-D177.txt" "S11073-D178.txt" "S11073-D179.txt" "S11073-D180.txt"
[173] "S11073-D182.txt" "S11073-D183.txt" "S11073-D184.txt" "S11073-D185.txt"
[177] "S11073-D186.txt" "S11073-D187.txt" "S11073-D188.txt" "S11073-D189.txt"
[181] "S11073-D190.txt" "S11073-D191.txt" "S11073-D192.txt" "S11073-D193.txt"
[185] "S11073-D194.txt" "S11073-D195.txt" "S11073-D196.txt" "S12296-D003.txt"
[189] "S12296-D005.txt" "S12296-D006.txt" "S12296-D007.txt" "S12296-D010.txt"
[193] "S12296-D011.txt" "S12296-D012.txt" "S12296-D013.txt" "S12296-D014.txt"
[197] "S12296-D015.txt" "S12296-D016.txt" "S12296-D017.txt" "S12296-D018.txt"
[201] "S12296-D019.txt" "S12296-D020.txt" "S12296-D021.txt" "S12296-D022.txt"
[205] "S12296-D023.txt" 

In [29]:
# Confirm working directory
getwd()

[1] "/Users/alaynemoody/Dropbox/Thesis/North_American_Letters_Diaries_OralHistories"

In [31]:
# Set working directory
setwd("letters")

In [36]:
# What are the files in that directory?
list.files()

[1] "S10003-D023.txt" "S10003-D024.txt" "S10003-D025.txt" "S10003-D026.txt"
   [5] "S10003-D027.txt" "S10003-D028.txt" "S10003-D029.txt" "S10003-D030.txt"
   [9] "S10003-D031.txt" "S10003-D032.txt" "S10003-D033.txt" "S10003-D034.txt"
  [13] "S10003-D035.txt" "S10003-D036.txt" "S10003-D037.txt" "S10003-D038.txt"
  [17] "S10003-D040.txt" "S10003-D041.txt" "S10003-D043.txt" "S10003-D044.txt"
  [21] "S10003-D046.txt" "S10003-D047.txt" "S10003-D048.txt" "S10003-D050.txt"
  [25] "S10003-D051.txt" "S10003-D052.txt" "S10003-D053.txt" "S10003-D054.txt"
  [29] "S10003-D055.txt" "S10003-D056.txt" "S10003-D057.txt" "S10003-D058.txt"
  [33] "S10003-D059.txt" "S10003-D060.txt" "S10003-D061.txt" "S10003-D062.txt"
  [37] "S10003-D063.txt" "S10003-D064.txt" "S10003-D065.txt" "S10003-D066.txt"
  [41] "S10003-D067.txt" "S10003-D068.txt" "S10003-D069.txt" "S10003-D072.txt"
  [45] "S10003-D074.txt" "S10003-D075.txt" "S10003-D081.txt" "S10003-D082.txt"
  [49] "S10003-D083.txt" "S10003-D084.txt" "S10003-D085.txt" "S10003-D086.txt"
  [53] "S10003-D088.txt" "S10003-D089.txt" "S10003-D091.txt" "S10003-D093.txt"
  [57] "S10003-D094.txt" "S10003-D096.txt" "S10003-D097.txt" "S10003-D098.txt"
  [61] "S10003-D100.txt" "S10003-D101.txt" "S10003-D102.txt" "S10003-D103.txt"
  [65] "S10003-D104.txt" "S10003-D105.txt" "S10003-D106.txt" "S10003-D108.txt"
  [69] "S10003-D109.txt" "S10003-D110.txt" "S10003-D111.txt" "S10003-D112.txt"
  [73] "S10003-D113.txt" "S10003-D114.txt" "S10003-D116.txt" "S10003-D117.txt"
  [77] "S10003-D118.txt" "S10003-D119.txt" "S10003-D120.txt" "S10003-D121.txt"
  [81] "S10003-D122.txt" "S10003-D123.txt" "S10003-D124.txt" "S10003-D125.txt"
  [85] "S10003-D126.txt" "S10003-D127.txt" "S10003-D128.txt" "S10003-D129.txt"
  [89] "S10003-D130.txt" "S10003-D134.txt" "S10003-D135.txt" "S10003-D136.txt"
  [93] "S10003-D137.txt" "S10003-D138.txt" "S10003-D139.txt" "S10003-D140.txt"
  [97] "S10003-D141.txt" "S10003-D142.txt" "S10003-D143.txt" "S10003-D144.txt"
 [101] "S10003-D145.txt" "S10003-D146.txt" "S10003-D147.txt" "S10003-D148.txt"
 [105] "S10003-D149.txt" "S10003-D150.txt" "S10003-D151.txt" "S10003-D152.txt"
 [109] "S10003-D153.txt" "S10003-D154.txt" "S10003-D155.txt" "S10003-D156.txt"
 [113] "S10003-D157.txt" "S10003-D158.txt" "S10003-D159.txt" "S10003-D160.txt"
 [117] "S10003-D161.txt" "S10003-D162.txt" "S10003-D163.txt" "S10003-D164.txt"
 [121] "S10003-D165.txt" "S10003-D166.txt" "S10003-D167.txt" "S10003-D168.txt"
 [125] "S10003-D169.txt" "S10003-D170.txt" "S10003-D171.txt" "S10003-D172.txt"
 [129] "S10003-D173.txt" "S10003-D174.txt" "S10003-D175.txt" "S10003-D176.txt"
 [133] "S10003-D177.txt" "S10003-D178.txt" "S10003-D179.txt" "S10003-D180.txt"
 [137] "S1019-D002.txt"  "S1019-D004.txt"  "S1019-D005.txt"  "S1019-D006.txt" 
 [141] "S1019-D007.txt"  "S1019-D008.txt"  "S1019-D009.txt"  "S1019-D010.txt" 
 [145] "S1019-D011.txt"  "S1019-D012.txt"  "S1019-D013.txt"  "S1019-D014.txt" 
 [149] "S1019-D015.txt"  "S1019-D016.txt"  "S1019-D017.txt"  "S1019-D018.txt" 
 [153] "S1019-D019.txt"  "S1019-D020.txt"  "S1019-D021.txt"  "S1019-D022.txt" 
 [157] "S1019-D023.txt"  "S1019-D024.txt"  "S1019-D025.txt"  "S1019-D026.txt" 
 [161] "S1019-D027.txt"  "S1019-D028.txt"  "S1019-D029.txt"  "S1019-D030.txt" 
 [165] "S1019-D031.txt"  "S1019-D032.txt"  "S1019-D033.txt"  "S1019-D034.txt" 
 [169] "S1019-D035.txt"  "S1019-D036.txt"  "S1019-D037.txt"  "S1019-D038.txt" 
 [173] "S1019-D039.txt"  "S1019-D040.txt"  "S1019-D041.txt"  "S1019-D042.txt" 
 [177] "S1019-D043.txt"  "S1019-D044.txt"  "S1019-D045.txt"  "S1019-D046.txt" 
 [181] "S1019-D047.txt"  "S1019-D048.txt"  "S1019-D049.txt"  "S1019-D050.txt" 
 [185] "S1019-D051.txt"  "S1019-D052.txt"  "S1019-D053.txt"  "S1019-D054.txt" 
 [189] "S1019-D055.txt"  "S1019-D056.txt"  "S1019-D057.txt"  "S1019-D058.txt" 
 [193] "S11067-D002.txt" "S11067-D004.txt" "S11067-D005.txt" "S11067-D008.txt"
 [197] "S11067-D010.txt" "S11067-D012.txt" "S11067-D014.txt" "S11067-D015.txt"
 [201] "S11067-D017.txt" "S11067-D018.txt" "S11067-D0

In [37]:
# How many files are there?
length(list.files())

[1] 1212

In [38]:
# Delete the fileids
sapply(fileids, unlink)

S11067-D063.txt S11067-D064.txt S11067-D065.txt S11067-D066.txt S11067-D067.txt 
              0               0               0               0               0 
S11067-D072.txt S11067-D073.txt S11067-D080.txt S11067-D082.txt S11067-D086.txt 
              0               0               0               0               0 
S11067-D087.txt S11067-D089.txt S11067-D092.txt S11067-D095.txt S11067-D100.txt 
              0               0               0               0               0 
S11067-D101.txt S11067-D102.txt S11067-D104.txt S11067-D109.txt S11067-D111.txt 
              0               0               0               0               0 
S11067-D113.txt S11068-D008.txt S11068-D016.txt S11068-D025.txt S11068-D034.txt 
              0               0               0               0               0 
S11068-D040.txt S11068-D043.txt S11068-D044.txt S11068-D045.txt S11068-D046.txt 
              0               0               0               0               0 
S11068-D047.txt S11068-D048.txt S11068-D057.txt S11068-D077.txt S11068-D078.txt 
              0               0               0               0               0 
S11068-D083.txt S11068-D085.txt S11068-D101.txt S11068-D104.txt S11073-D008.txt 
              0               0               0               0               0 
S11073-D009.txt S11073-D010.txt S11073-D011.txt S11073-D012.txt S11073-D013.txt 
              0               0               0               0               0 
S11073-D014.txt S11073-D016.txt S11073-D018.txt S11073-D019.txt S11073-D020.txt 
              0               0               0               0               0 
S11073-D022.txt S11073-D023.txt S11073-D024.txt S11073-D026.txt S11073-D029.txt 
              0               0               0               0               0 
S11073-D030.txt S11073-D031.txt S11073-D032.txt S11073-D034.txt S11073-D035.txt 
              0               0               0               0               0 
S11073-D036.txt S11073-D037.txt S11073-D038.txt S11073-D039.txt S11073-D041.txt 
              0               0               0               0               0 
S11073-D043.txt S11073-D044.txt S11073-D045.txt S11073-D047.txt S11073-D048.txt 
              0               0               0               0               0 
S11073-D050.txt S11073-D051.txt S11073-D053.txt S11073-D054.txt S11073-D056.txt 
              0               0               0               0               0 
S11073-D057.txt S11073-D059.txt S11073-D061.txt S11073-D063.txt S11073-D064.txt 
              0               0               0               0               0 
S11073-D065.txt S11073-D067.txt S11073-D068.txt S11073-D069.txt S11073-D070.txt 
              0               0               0               0               0 
S11073-D071.txt S11073-D072.txt S11073-D073.txt S11073-D074.txt S11073-D076.txt 
              0               0               0               0               0 
S11073-D078.txt S11073-D079.txt S11073-D080.txt S11073-D083.txt S11073-D084.txt 
              0               0               0               0               0 
S11073-D085.txt S11073-D086.txt S11073-D087.txt S11073-D088.txt S11073-D090.txt 
              0               0               0               0               0 
S11073-D091.txt S11073-D093.txt S11073-D094.txt S11073-D095.txt S11073-D097.txt 
              0               0               0               0               0 
S11073-D099.txt S11073-D100.txt S11073-D101.txt S11073-D103.txt S11073-D104.txt 
              0               0               0               0               0 
S11073-D105.txt S11073-D107.txt S11073-D109.txt S11073-D110.txt S11073-D112.txt 
              0               0               0               0               0 
S11073-D113.txt S11073-D116.txt S11073-D117.txt S11073-D119.txt S11073-D120.txt 
              0               0               0               0               0 
S11073-D122.txt S11073-D123.txt S11073-D125.txt S11073-D126.txt S11073-D127.txt 
              0               0

[1] 925

In [39]:
# Check it worked correctly by getting count.
length(list.files())

[1] 925

In [42]:
# Converts character to factor class
factorize <- function(df){
  for(i in which(sapply(df, class) == "character")) df[[i]] = as.factor(df[[i]])
  return(df)
}

In [43]:
# Refactorize the data
lettersFinal  <-  factorize(lettersFinal)
glimpse(lettersFinal)

Observations: 925
Variables: 32
$ X                       <int> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14…
$ docsequence             <int> 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 3…
$ docid                   <fct> S10003-D023, S10003-D024, S10003-D025, S10003…
$ docyear                 <int> 1836, 1836, 1837, 1837, 1838, 1838, 1838, 183…
$ doctype                 <fct> Letter, Letter, Letter, Letter, Letter, Lette…
$ allsubject              <fct> "Childbirth; Church attendance; Cities; Farms…
$ broadsubj               <fct> Health; Religion; Communities; Relationships;…
$ personalevent           <fct> NA, NA, NA, NA, NA, NA, NA, NA, Physical illn…
$ wwritten                <fct> "Baltimore, MD; Maryland; United States; Mid-…
$ docauthorid             <fct> per0022938, per0022938, per0022938, per002293…
$ docauthorname           <fct> "Bruns, Jette, 1813-1899", "Bruns, Jette, 181…
$ language                <fct> English; German, English; German, English; Ge…
$ editor            

In [45]:
# Write a new .csv
write.csv(lettersFinal, "../20201125_AM_Meta2Merge.csv")